include/zephyr/sys/util.h

```diff
@@ -271,6 +271,16 @@ extern "C" {
 		((type *)(((char *)(ptr)) - offsetof(type, field))); \
 	})
 
+/**
+ * @brief Report the size of a struct field in bytes.
+ *
+ * @param type The structure containing the field of interest.
+ * @param member The field to return the size of.
+ *
+ * @return The field size.
+ */
+#define sizeof_field(type, member) sizeof((((type *)0)->member))
```

Comment:
I give even odds that some static analyzer or MISRA checker is going to flag this, but... not my monkeys, as it were.

Reply:
Thank for the review, I changed the name to all caps.
Indeed, maybe one more checking rule is needed to avoid this issue.